xarray/conventions.py

```diff
@@ -776,6 +815,10 @@ def decode_cf_variable(var, concat_characters=True, mask_and_scale=True,
     else:
         encoding['dtype'] = original_dtype
 
+    if 'dtype' in attributes and attributes['dtype'] == 'bool':
+        del attributes['dtype']
```

Comment:
@shoyer - do you think it makes since to throw this attribute away? The roundtrip fails in its current state if I move it to encoding or leave it where it is.